lib/bap_sema/bap_sema_lift.ml

```diff
@@ -192,61 +201,66 @@ module IrBuilder = struct
       ?is_barrier:bool ->
       ?fall:[> `Inter of Ir_jmp.dst | `Intra of Ir_jmp.dst ] ->
       ?addr:addr ->
-      blk term list -> blk term list =
+      blk term list -> blk term list KB.t =
     fun ?(is_call=false) ?(is_barrier=false) ?fall ?addr blks ->
     let blks = with_landing_pads fall blks in
-    with_first_blk_optionally_addressed addr @@
-    concat_map_fst_and_rev blks @@ function
-    | x when is_barrier -> [x]
-    | x -> match fall with
-      | None -> [if is_call || has_call x then turn_into_call None x else x]
-      | Some (`Intra dst) -> [
-          fall_if_possible dst @@
-          if is_call || has_call x
-          then turn_into_call (Some dst) x
-          else x
-        ]
-      | Some (`Inter dst) ->
-        if is_call || has_call x then
-          let next = Ir_blk.create () in
-          let fall = Ir_jmp.resolved (Term.tid next) in
-          let next = insert_inter_fall dst next in [
-            next;
-            fall_if_possible fall @@
-            turn_into_call (Some fall) x
+    begin concat_map_fst_and_rev blks @@ function
```

Comment:
Those begin/end blocks make the code really unreadable. It was already asking for some refactoring so could you please refactor them into functions?